### PR TITLE
fix users had already joined a workspace, but the system still first …

### LINKED ISSUE #9833: OAuth Login Error for Users with Existing Workspaces After Upgrade to Version 0.10.1
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.10.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

After upgrading the project from version 0.6.14 to 0.10.1, users who previously logged in via OAuth are encountering the following error upon logging in again:`Workspace not found, please contact system admin to invite you to join in a workspace.`

1. Upgrade the project from version 0.6.14 to 0.10.1.
2. Log in using OAuth with a user account that has already joined a workspace.
3. Attempt to log in again.
image

### ✔️ Expected Behavior

Users who have already joined a workspace should be able to log in without encountering an error message about the workspace not being found.

### ❌ Actual Behavior

This issue affects all users who have previously logged in via OAuth and have already joined a workspace. Fixing this will improve the user experience and reduce confusion.

### PR DESCRIPTION
**Pull request by G81192**
### Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I ran `dev/reformat` (backend) and `cd web && npx lint-staged` (frontend) to appease the lint gods.

### Description

When updating the project from version 0.6.14 to 0.10.1, users who previously logged in via OAuth encountered the error "Workspace not found, please contact system admin to invite you to join in a workspace" upon logging in again. These users have already joined a workspace, but the system first checks if they can create their own workspace, leading to login failures when this check fails. I believe this sequence is incorrect. If a user is already linked to a workspace, it should return immediately without checking `is_allow_create_workspace`. This PR aims to fix this bug.

Fixes #9833

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

### Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/services/account_service.py [hunk 1]
```diff
@@ -505,16 +505,19 @@ def create_tenant(name: str, is_setup: Optional[bool] = False) -> Tenant:
     def create_owner_tenant_if_not_exist(
         account: Account, name: Optional[str] = None, is_setup: Optional[bool] = False
     ):
-        """Create owner tenant if not exist"""
```
Comment (crazywoola) [inline api/services/account_service.py:508]: I don't think this should be removed.

Comment (G81192) [reply api/services/account_service.py:508]: > I don't think this should be removed.

Thank you for your feedback, @crazywoola . Just to clarify, I’m not suggesting the remove it. Instead, I propose moving the functionality that checks if a user has a workspace to occur before the is_allow_create_workspace validation. If the user already has a workspace, we should bypass the check for whether they can create a new one. This change aims to streamline the login process and prevent unnecessary login failures for users with existing workspaces. I appreciate your insights and am open to further discussion on this approach.

Comment (crazywoola) [reply api/services/account_service.py:508]: I may have forgotten some of this code, maybe @ZhouhaoJiang can help you with more insights.

**Review by ZhouhaoJiang — APPROVED** [review]
LGTM

**Review by ZhouhaoJiang — COMMENTED** [review]
Please check lint

**Review by ZhouhaoJiang — CHANGES_REQUESTED** [review]
run `dev/reformat`

**Comment by G81192** [comment]
> run `dev/reformat`

@ZhouhaoJiang Thank you for your feedback! I have executed the dev/reformat command to resolve any linting issues. Please let me know if there's anything else that needs to be addressed.

**Review by ZhouhaoJiang — APPROVED** [review]
LGTM

**Pull request merged**